### PR TITLE
allow exporting of projects to a CSV file

### PR DESCRIPTION
**Pull request by jonmoter**
I wanted to download a CSV file of all the projects in Samson, along with their owners, so we could see which projects belonged to which teams, and so we can easily export the set of projects into a spreadsheet.

/cc @zendesk/samson, @pswadi-zendesk, @JLandersZen, @kazraza 

### Tasks
 - [ ] :+1: from team

### Risks
- Level: Low

### CODE REVIEW

**Comment by irwaters** [comment]
looks like a rubocop nag.

File: app/serializers/project_serializer.rb [hunk 1]
```diff
@@ -1,8 +1,16 @@
 # frozen_string_literal: true
 class ProjectSerializer < ActiveModel::Serializer
-  attributes :id, :name, :url, :permalink, :repository_url
+  attributes :id, :name, :url, :permalink, :repository_url, :owner, :created_at
+
+  def self.csv_header
```
Comment (grosser) [inline app/serializers/project_serializer.rb]: this relies on the ordering of attributes and csv_line to be always the same, which is not tested ... how about 
```
PROPERTIES = [:id, :foo]
attributes *PROPERTIES

def csv_header
  PROPERTIES.map { |p| p.to_s.humanize }
end

def csv_line
  PROPERTIES.each { |p| public_send p }
end
```

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -20,6 +22,11 @@ def index
       format.json do
         render json: Project.ordered_for_user(current_user).all
       end
+
+      format.csv do
+        datetime = Time.now.strftime "%Y%m%d_%H%M"
```
Comment (grosser) [inline app/controllers/projects_controller.rb]: how about 2017-02-01_13-34

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -100,4 +107,16 @@ def projects_for_user
   def require_project
     @project = (Project.find_by_param!(params[:id]) if params[:id])
   end
+
+  def as_csv
+    projects = Project.order(:id).all
```
Comment (grosser) [inline app/controllers/projects_controller.rb]: `:id` is the default anyway ?

File: app/controllers/projects_controller.rb (hunk 2 above)
Comment (jonmoter) [inline app/controllers/projects_controller.rb]: Okay.  Copy-pasted from the other place we're exporting a CSV.

File: app/serializers/project_serializer.rb (hunk 1 above)
Comment (jonmoter) [inline app/serializers/project_serializer.rb]: `csv_line` should be `PROPERTIES.map`, right?

File: app/controllers/projects_controller.rb (hunk 3 above)
Comment (jonmoter) [inline app/controllers/projects_controller.rb]: Being explicit about it, since there's a controller test that assumes it's ordered by ID.

**Comment by jonmoter** [comment]
Updated based on feedback.

**Pull request merged**